### DISCUSSION TITLE
"Yourself" room

### DISCUSSION THREAD

**Discussion by lukebarnard1** [element-hq/element-meta#739] [Ideas]
Following on from vector-im/element-web#4833, it was suggested that Riot has a "Start a chat with yourself" feature that creates a new "DM" with just you in it, called "Yourself".

@Nostradamos [gave a description](https://github.com/vector-im/riot-web/issues/4833#issuecomment-324500385):
> Telegram has the feature to chat with "yourself", i like the idea because it can be used as a clipboard across multiple devices, quickly sending a picture/url/whatever from your smartphone to your desktop is a nice thing to have. All of this works if you're in an empty room but it would be nice to have it act a bit different:
> 
> - be able to start a chat with yourself without the need of creating an empty room first
> - Show a chat with yourself under "Persons" and not list it as a room.
> - directly set the name to something meaningful like "Yourself" or your own nickname
> 

This could be quite a nice feature. In terms of implementing, do we want to

1. change the semantics of being in a room alone? OR
1. provide the ability to create a new room and explicitly setting the room name to "Yourself" in the chosen language and putting it under "People"?

The negative of 2. is that someone might chose a language, create a room with "Yourself" and then change to another language.

Changing the semantics of being in a room alone means changing "Empty Room" to "Yourself" (and this is generated when the room has no name, so there is no i18n problem.) However, it would be quite strange to move "Empty Room"s into "People", and I think it could be quite confusing to see a room with you and one other person in it become a direct message. Seeing the name changed to "Yourself" would probably be fine though.

**Comment by turt2live** [comment]
The awkward part of the name changing from "Empty Room" to "Yourself" would be the case of a previous 1:1 with someone and they have left the room. The room's avatar is still left as that person, and seeing "Yourself" next to their avatar would be misleading/confusing.

**Comment by t3chguy** [comment]
@turt2live you could make the case of
`currentState['m.room.member'].length === 1` then call it "Yourself"
that way if someone joins and leaves it'll go Empty Room instead

**Comment by lampholder** [comment]
I guess the fundamentals of this feature exist today, it's just a matter of how we dress it up so that people realise how/why they can use Riot.im in this way.

**Comment by lampholder** [comment]
Thoughts on this:

Slack:
- lists your own name in the list of "Direct Messages"
- changes the composer placeholder text from 'Message #channel' or 'Message @user' to 'Jot something down'
- replaces the 'This is the very beginning of your direct messge history with *name*' text with a description of how you might use your channel talking to yourself.
- lets you leave your chat with yourself, but if you initiate a new one in has the original chat contents in it

Of course, being Slack, you're limited to just the one room with yourself :(

We might consider:
- creating a DM chat for users with themselves by default 
- adding some hints as to how they might want to use this room

> The awkward part of the name changing from "Empty Room" to "Yourself" would be the case of a previous 1:1 with someone and they have left the room. The room's avatar is still left as that person, and seeing "Yourself" next to their avatar would be misleading/confusing.

I didn't realise this was the behaviour - I don't think I like it :P If I'm chatting with X and then X leaves, how am I meant to find those message logs? I'd filter the chat list for X and wouldn't find it because the room is now called 'Empty Room'.

Possibilities to consider:
- getting rid of 'Empty Room' (there's no such thing; it literally always has at least you in it) and replacing with perhaps 'New Room'
- perhaps modifying the room auto-naming behaviour
    - perhaps making it only incremental
    - perhaps not summarising the participants-based generated name to 'X and 2 others'
    - perhaps just making it so the last person to leave doesn't take their name away from the room
    - if we do want the last person to leave to take away their name, make it also take away their avatar
- perhaps modifying the room filter so that it shows you both rooms with names that match your filter term _and_ DM rooms with matching participants (current or past)

**Comment by Nostradamos** [comment]
> I didn't realise this was the behaviour - I don't think I like it :P If I'm chatting with X and then X leaves, how am I meant to find those message logs? I'd filter the chat list for X and wouldn't find it because the room is now called 'Empty Room'.

I would second that. This behaviour doesn't make any sense, room name should always stay the same in my opinion.

> provide the ability to create a new room and explicitly setting the room name to "Yourself" in the chosen language and putting it under "People"?

I think this is maybe the best solution, but of course i see your point with changing the language. Is it somehow possible to differently flag the chat? We could also just name it `yourself` and map the translated name over it in riot, but this would obviously be some "riot magic".

> creating a DM chat for users with themselves by default

This is in my opinion also not the best solution, because not anyone is using this feature, and creating just another room for every user is maybe also polluting the homeserver a bit.

**Comment by Leon0402** [comment]
I also like about Telegram that it's there by default and is always at the top with it's own icon. In Telegram it's called "Saved Messages" that seems like a good name to me. 
"Yourself" seems like an odd name to me.

**Comment by hex-m** [comment]
This works for me.

1. create new room
1. Name: "Note to self" (or whatever you like)
1. click "stop warning about empty room".

The Rooms stays in the "Rooms" category and not in "People" which may be confusing. But there is vector-im/element-web#8614 that is about allowing users to put a Room where they want it.

So this got solved?

**Comment by Ekleog** [comment]
It would be awesome to have something like this, and to have such a room always easily accessible (eg. next to the display name in the top-left corner, to not take too much space?).

It would make it possible to always have a “no-op” room to switch to when done reading element without having to close the tab. Otherwise, one has to always leave element on one chat room, which makes it much too easy to mark messages as read, especially until vector-im/element-web#15726, vector-im/element-web#6386 and vector-im/element-meta#347 are all done. (Even though I think there would still be a use for such an easily accessible “stop marking messages in this room as read” mode even with all these fixed)

**Comment by oghaki** [comment]
> This works for me.
> 
> 1. create new room
> 2. Name: "Note to self" (or whatever you like)
> 3. click "stop warning about empty room".
> 
> The Rooms stays in the "Rooms" category and not in "People" which may be confusing. But there is vector-im/element-web#8614 that is about allowing users to put a Room where they want it.
> 
> So this got solved?

I think this option would be perfect.  I use this feature all the time in Telegram, and I did want to point out a few considerations related to how I use it, which might help when designing the implementation.  

1. I use this feature for two main purposes:
     i. To quickly send links, media, etc. either between devices or almost like a bookmark, to recover soon after. 
     ii. To store information long-term that I can search and reference later (I’ll usually add a specific string of text to act as a kind of tagging to the message, so that it is easier to search for in the future).  
2. The room should be special (e.g., a ‘note-to-self’ type, or similar, as mentioned), not indistinguishable from any room which has no other participants.  This is to facilitate easy and unambiguous access to it, so that chats are sent in, and the chat history remains in, one place.  With that said, I like the idea of being able to create multiple instances, so that one could classify information based on which chat it was sent in (taking the place of the strings I add in Telegram for tagging).
3. It would also be nice to make the interface for the room a bit more compact by default, as you cannot call or video-chat yourself, and there may be other features that can be removed from the UI to limit clutter and allow the user to reduce the window size. 

Thus far, I’ve found two different ways to create this type of chat—first, I just created a room private to myself and named it ‘Saved Messages’, which appears under ‘Rooms’, and next, I just started a new chat with myself, which appears under ‘People’

**Comment by kittykat** [comment]
I use a DM with myself often for many of the above reasons.

Where there is a "start a DM" link in another user's profile, this could be a "start a DM with yourself" which could lead to a room that is a DM type and is titled with your own username.

**Comment by hex-m** [comment]
So we mostly agree this would be nice. Any chance to get in included in a roadmap or have the product team look into the idea?

**Comment by Mikaela** [comment]
Wouldn't this be an aspect of canonical DMs https://github.com/matrix-org/matrix-spec-proposals/pull/2199 instead of only Elements altering themselves?

I think the desired effect can be reached by creating a new room and then `/converttodm` in the meantime
